File: Practicas/Yahir-Hermosillo/Parcial3/Practica1/CreditCardPayment.java
class CreditCardPayment extends PaymentMethod {
    private String cardNumber;
    private String expirationDate;
    private String cvv;
    
    public CreditCardPayment(String cardNumber, String expirationDate, String cvv) {
        this.cardNumber = cardNumber;
        this.expirationDate = expirationDate;
        this.cvv = cvv;
    }
    
    @Override
    public void processPayment(double amount) {
        System.out.printf("Procesando pago con tarjeta de $%.2f - Número: ****%s%n", 
                         amount, cardNumber.substring(cardNumber.length() - 4));
    }
}

File: Practicas/Marco-Oregel/Practica1/CreditCardPayment.py
from PaymentMethod import PaymentMethod

class CreditCardPayment(PaymentMethod):

    def __init__(self,number,name,date,cvv):
        self._number = number
        self._name = name
        self._date = date
        self._cvv = cvv

    def processPayment(self,amount):
        print(f"Se ha procesado su pago de: $ {amount} pesos con tarjeta de credito de: {self._name}")
